src/ripple/rpc/handlers/ServerInfo.cpp

```diff
@@ -22,13 +22,308 @@
+            v[jss::isVLEncoded] = false;
+            v[jss::isSerialized] = false;
+            v[jss::isSigningField] = false;
+            v[jss::type] = "Amount";
+            a[1U] = v;
+            ret[jss::FIELDS][i++] = a;
+        }
+
+        for (auto const& [code, f] : ripple::SField::knownCodeToField)
+        {
+            if (f->fieldName == "")
+                continue;
+
+            Json::Value innerObj = Json::objectValue;
+
+            uint32_t type = f->fieldType;
+
+            innerObj[jss::nth] = f->fieldValue;
+
+            innerObj[jss::isVLEncoded] =
+                (type == 7U /* Blob       */ || type == 8U /* AccountID  */ ||
+                 type == 19U /* Vector256  */);
+
+            innerObj[jss::isSerialized] =
+                (type < 10000 && f->fieldName != "hash" &&
+                 f->fieldName !=
+                     "index"); /* hash, index, TRANSACTION, LEDGER_ENTRY,
+                                  VALIDATION, METADATA */
+
+            innerObj[jss::isSigningField] = f->shouldInclude(false);
+
+            innerObj[jss::type] = type_map[type];
+
+            Json::Value innerArray = Json::arrayValue;
+            innerArray[0U] = f->fieldName;
+            innerArray[1U] = innerObj;
+
+            ret[jss::FIELDS][i++] = innerArray;
+        }
+
+        ret[jss::TRANSACTION_RESULTS] = Json::objectValue;
+
+        for (const auto& pair : detail::transResults())
+        {
+            ret[jss::TRANSACTION_RESULTS][str(pair.second.first)] =
+                std::stoi(str(pair.first));
+        }
+
+        ret[jss::TRANSACTION_TYPES] = Json::objectValue;
+        ret[jss::TRANSACTION_TYPES][jss::Invalid] = -1;
+        for (auto it = TxFormats::getInstance().begin();
+             it != TxFormats::getInstance().end();
+             ++it)
+        {
+            ret[jss::TRANSACTION_TYPES][str(it->getName())] =
+                std::stoi(str(it->getType()));
+        }
+
+        // generate hash
+        {
+            const std::string out = Json::FastWriter().write(ret);
+            defsHash =
+                ripple::sha512Half(ripple::Slice{out.data(), out.size()});
+            ret[jss::hash] = to_string(*defsHash);
+        }
+
+        return ret;
+    }
+
+    std::optional<uint256> defsHash;
+    Json::Value defs;
+
+public:
+    Definitions() : defs(generate()){};
+
+    bool
+    hashMatches(uint256 hash) const
+    {
+        return defsHash && *defsHash == hash;
+    }
+
+    Json::Value const&
+    operator()(void) const
+    {
+        return defs;
+    }
+};
+
+Json::Value
+doServerDefinitions(RPC::JsonContext& context)
+{
+    auto& params = context.params;
+
+    uint256 hash;
+    if (params.isMember(jss::hash))
+    {
+        if (!params[jss::hash].isString() ||
+            !hash.parseHex(params[jss::hash].asString()))
+            return RPC::invalid_field_error(jss::hash);
+    }
+
+    static const Definitions defs{};
+    if (defs.hashMatches(hash))
```

Comment:
The value of `hash` is default-constructed on line 309. How does it represent a user's input?

Reply:
https://github.com/XRPLF/rippled/pull/4703#discussion_r1330641082

Reply:
Okay. I still don't understand how `hash` captures the user input? `defs` refer to the definitions contained by the server. Shouldn't we read the user input from `params.context` ?

Reply:
That's exactly what's happening here - on line 313.

Reply:
i completely missed that, understood 👍